paraatm/simulation_method/aviationr.py

```diff
@@ -0,0 +1,210 @@
+"""
+NASA NextGen NAS ULI Information Fusion
+
+@organization: Arizona State University
+@author: Xinyu Zhao
+@date: 2020-05-30
+@last updated: 2020-05-30
+
+This Python script is used for simulating aviation accident based on recording in NTSB
+Simulation based on NATS beta1.7 standalone version
+"""
+
+import os
+import pandas as pd
+from paraatm.simulation_method.aviationr_model import RiskEstimator
+
+from paraatm.io.nats import NatsSimulationWrapper, NatsEnvironment
+
+
+class AviationRisk(NatsSimulationWrapper, object):
+    """Class method for aviation accident simulation and quantifying risk
+    """
+    def __init__(self, cfg):
+        """
+
+        Parameters
+        ----------
+        cfg : dictionary
+            inputs to Aviation Risk simulation
+            
+            'fp_file' :  str
+                directory tp flight plan file for NATS simulation
+            'mfl_file' : str
+                directory to mfl file for NATS simulation
+            'data_file' : str
+                required data for accident simulation and risk estimation
+            'model_file' : srt
+                directory to pre-trained model
+            'sim_time' : int/float
+                total simulation duration
+        """
+        self.fp_file = cfg['fp_file']
+        self.mfl_file = cfg['mfl_file']
+        self.data = cfg['data_file']
+        self.case = self.data + 'case.pickle'
+        self.refer = self.data + 'refer.pickle'
+        self.model = cfg['model_file']
+        self.sim_time = cfg['sim_time']
+
+
+        NatsEnvironment.start_jvm(nats_home=None)
+        self.NATS_SIMULATION_STATUS_PAUSE = NatsEnvironment.get_nats_constant('NATS_SIMULATION_STATUS_PAUSE')
+        self.NATS_SIMULATION_STATUS_ENDED = NatsEnvironment.get_nats_constant('NATS_SIMULATION_STATUS_ENDED')
+
+        natsStandalone = NatsEnvironment.get_nats_standalone()
+
+        self.simulationInterface = natsStandalone.getSimulationInterface()
+
+        self.entityInterface = natsStandalone.getEntityInterface()
+        self.controllerInterface = self.entityInterface.getControllerInterface()
+        self.pilotInterface = self.entityInterface.getPilotInterface()
+
+        self.environmentInterface = natsStandalone.getEnvironmentInterface()
+
+        self.equipmentInterface = natsStandalone.getEquipmentInterface()
+        self.aircraftInterface = self.equipmentInterface.getAircraftInterface()
+
+        if self.simulationInterface is None:
+            natsStandalone.stop()
+            raise RuntimeError("Can't get SimulationInterface")
+
+        self.simulationInterface.clear_trajectory()
+
+    def simulation(self, device, delay=1000, isRNN=True):
+        """
+        Main function for simulation. NATS simulation based commands
+
+        Parameters
+        ----------
+        device : str
+           'CUDA'/ 'CPU'-- Running the model along GPU or CPU
```

Comment:
I don't think this is correct.  I think the argument you are using is actually a `torch.device` instance (not a str).

Also, you might consider providing a default value of `None` and then doing:
```python
if device is None:
    device = torch.device('cpu')
    if torch.cuda.is_available():
        device = torch.device('cuda')
```
If you want the user to pass a str instead of a `torch.device` instance, you would need to add an "else" clause with `device = torch.device(device)`.

Reply:
Or actually, it looks like passing a `str` does work, sorry about that.  You might still consider providing a default value handling.

Reply:
Got it. I have changed the type into torch.device and add the default value.